pennylane/qchem/vibrational/taylor_ham.py

```diff
@@ -0,0 +1,594 @@
+        ).normal_order()
+        harm_pot += bosonized_qm2 * freqs[mode] * 0.5
+
+    return harm_pot.normal_order()
+
+
+def taylor_bosonic(taylor_coeffs_array, freqs, is_local=True, uloc=None):
+    """Return Taylor bosonic vibrational Hamiltonian.
+
+     The construction of the Hamiltonian is based on Eqs. 4-7 of `arXiv:1703.09313 <https://arxiv.org/abs/1703.09313>`_.
+
+    Args:
+        taylor_coeffs (list(float)): the coefficients of the Hamiltonian
+        freqs (list(float)): the harmonic frequencies in reciprocal centimetre
+        is_localal (bool): Flag whether the vibrational modes are localized. Default is ``True``.
+        uloc (list(float)): localization matrix indicating the relationship between original and
+            localized modes
+
+    Returns:
+        BoseSentence: Taylor bosonic hamiltonian
+    """
+    if is_local:
+        start_deg = 2
+    else:
+        start_deg = 3
+
+    harm_pot = _taylor_harmonic(taylor_coeffs_array, freqs)
+    ham = _taylor_anharmonic(taylor_coeffs_array, start_deg) + harm_pot
+    kin_ham = _taylor_kinetic(taylor_coeffs_array, freqs, is_local, uloc)
+    ham += kin_ham
+    return ham.normal_order()
+
+
+def taylor_hamiltonian(
+    pes, max_deg=4, min_deg=3, mapping="binary", n_states=2, ps=False, wire_map=None, tol=None
+):
+    """Return Taylor vibrational Hamiltonian.
+
+    Args:
+        pes (VibrationalPES): object containing the vibrational potential energy surface data
+        max_deg (int): maximum degree of Taylor form polynomial
+        min_deg (int): minimum degree of Taylor form polynomial
+        mapping (str): Mapping used to map to qubit basis. Default is `"binary"`.
+        n_states(int): maximum number of allowed bosonic states
+        ps (bool): Whether to return the result as a PauliSentence instead of an
+            operator. Defaults to False.
+        wire_map (dict): A dictionary defining how to map the states of
+            the Bose operator to qubit wires. If None, integers used to
+            label the bosonic states will be used as wire labels. Defaults to None.
+        tol (float): tolerance for discarding the imaginary part of the coefficients
+
+    Returns:
+        BoseSentence: the bosonic form of the Taylor Hamiltonian
+    """
+    mapping.lower().strip()
+    coeffs_arr = taylor_coeffs(pes, max_deg, min_deg)
+    bose_op = taylor_bosonic(coeffs_arr, pes.freqs, is_local=pes.localized, uloc=pes.uloc)
+    if mapping == "binary":
+        ham = binary_mapping(
+            bose_operator=bose_op, n_states=n_states, ps=ps, wire_map=wire_map, tol=tol
+        )
+    elif mapping == "unary":
+        ham = unary_mapping(
+            bose_operator=bose_op, n_states=n_states, ps=ps, wire_map=wire_map, tol=tol
+        )
+    else:
+        raise ValueError(
+            f"Specified mapping {mapping}, is not found. Please use either 'binary' or 'unary' mapping."
+        )
```

Comment:
We should check this in the beginning to avoid going through the whole function to raise an error.